libs/cmdio/io.go

```diff
@@ -88,6 +88,29 @@ func (c *cmdIO) IsTTY() bool {
 	return isatty.IsTerminal(fd) || isatty.IsCygwinTerminal(fd)
 }
 
+func IsPromptSupported(ctx context.Context) bool {
+	// We do not allow prompting in non-interactive mode and in Git Bash.
+	// Likely due to fact that Git Bash does not (correctly support ANSI escape sequences,
+	// we cannot use promptui package there.
+	// See known issues:
+	// - https://github.com/manifoldco/promptui/issues/208
+	// - https://github.com/chzyer/readline/issues/191
+	// We also do not allow prompting in non-interactive mode,
+	// because it's not possible to read from stdin in non-interactive mode.
+	return IsInteractive(ctx) && IsOutTTY(ctx) && IsInTTY(ctx) && !IsGitBash(ctx)
+}
+
+func IsGitBash(ctx context.Context) bool {
+	// Check if the MSYSTEM environment variable is set to "MINGW64"
+	msystem := os.Getenv("MSYSTEM")
+	if strings.EqualFold(msystem, "MINGW64") {
+		// Check for typical Git Bash env variable for prompts
+		return os.Getenv("PS1") != ""
+	}
```

Comment:
Please use `env.Get` on the context, then this can be tested without actually setting env vars.